### PR TITLE
1.20.2 update

### PR DESCRIPTION
**Pull request by Tides**
Update for 1.20.2. Lots of changes to the client login sequence 

Any changes made in the Obsidian.Net.Packets namespace can be ignored cause its just all packet ids/a couple new packets

### CODE REVIEW

File: Obsidian/Client.cs [hunk 1]
```diff
@@ -469,34 +500,28 @@ private async Task HandleEncryptionResponseAsync(byte[] data)
...
-        //await this.SetCompression();
-        await ConnectAsync();
+        this.SendPacket(new LoginSuccess(Player.Uuid, Player.Username)
+        {
+            SkinProperties = this.Player.SkinProperties,
+        });
     }
 
-    // TODO fix compression (.net 6)
+    // TODO fix compression now????
```
Comment (Naamloos) [inline Obsidian/Client.cs:509]: Do we have a GH issue open for this?

File: Obsidian/Client.cs [hunk 2]
```diff
@@ -607,27 +629,15 @@ internal void SendKeepAlive(DateTimeOffset time)
             return;
         }
 
-        Logger.LogDebug($"Doing KeepAlive ({keepAliveId}) with {Player.Username} ({Player.Uuid})");
+        Logger.LogDebug("Doing KeepAlive ({keepAliveId}) with {Username} ({Uuid})", keepAliveId, Player.Username, Player.Uuid);
```
Comment (Naamloos) [inline Obsidian/Client.cs:632]: We should look into other uses of the logger that don't use string formatting in this way, but that's out of scope for this PR.

File: Obsidian/Net/Packets/ClientState.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@ public enum ClientState
     Handshaking = 0,
     Status = 1,
     Login = 2,
-    Play = 3,
+    Configuration = 3,
```
Comment (Naamloos) [inline Obsidian/Net/Packets/ClientState.cs:8]: cool that they added a config state

File: Obsidian/Net/Packets/Play/Clientbound/PlayerChatMessagePacket.cs [hunk 4]
```diff
@@ -3,21 +3,22 @@
 
 namespace Obsidian.Net.Packets.Play.Clientbound;
 
+//TODO this changed implement later
```
Comment (Naamloos) [inline Obsidian/Net/Packets/Play/Clientbound/PlayerChatMessagePacket.cs:6]: We might also want to open an issue for this to track progress.

File: Obsidian/Client.cs (hunk 1 above)
Comment (Tides) [reply Obsidian/Client.cs:509]: That's a good idea we should open an issue for it

File: Obsidian/Net/Packets/Play/Clientbound/UpdateAdvancementsPacket.cs [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 namespace Obsidian.Net.Packets.Play.Clientbound;
 
-//TODO finish this packet
+//TODO finish this packet PLEASE
```
Comment (Naamloos) [inline Obsidian/Net/Packets/Play/Clientbound/UpdateAdvancementsPacket.cs:6]: teehee

**Review by Naamloos — APPROVED** [review]
Nice work as always. As stated in the previous comments we might want to open GitHub issues for the TODOs scattered through the project. This may also be useful for new contributors wanting to help us develop the server further.

**Pull request merged**